File: lista_1/c_library/test_recursive_lib.c
#include <stdio.h>
#include "recursive_lib.h"
#include "struct_lib.h"

int main() {
    printf("Biblioteka rekurencyjna \n\n");
    unsigned int n;
    printf("Podaj liczbe naturalna do obliczenia silni: ");
    scanf("%d", &n);
    printf("%d! = %lld\n", n, factorial_recursive(n));

    unsigned int a, b;
    printf("\nPodaj dwie liczby naturalne do obliczenia NWD: ");
    scanf("%u %u", &a, &b);
    printf("NWD(%u, %u) = %u\n", a, b, gcd_recursive(a, b));

    int x, y, c;
    printf("\nPodaj wspolczynniki a, b i c dla rownania diofantycznego ax + by = c: ");
    scanf("%d %d %d", &a, &b, &c);
    Solution sol = diophantine_recursive(a, b, c);
    printf("Error: %s\n", sol.err ? "true" : "false");
    printf("Rozwiazanie: x = %d, y = %d\n", sol.x, sol.y);

    return 0;
}

File: lista_1/library_python/test_recursive_lib.py
# test_recursive_lib.py
from recursive_lib import factorial_recursive, gcd_recursive, diophantine_recursive

print("Biblioteka rekurencyjna\n")

n = int(input("Podaj liczbę naturalną do obliczenia silni: "))
print(f"{n}! = {factorial_recursive(n)}")

a, b = map(int, input("\nPodaj dwie liczby naturalne do obliczenia NWD: ").split())
print(f"NWD({a}, {b}) = {gcd_recursive(a, b)}")

a, b, c = map(int, input("\nPodaj współczynniki a, b i c dla równania diofantycznego ax + by = c: ").split())
sol = diophantine_recursive(a, b, c)
print(f"Error: {sol.err}")
print(f"Rozwiązanie: x = {sol.x}, y = {sol.y}")
